sentry-android-core/src/main/java/io/sentry/android/core/UserInteractionIntegration.java

```diff
@@ -0,0 +1,183 @@
+package io.sentry.android.core;
+
+import android.app.Activity;
+import android.app.Application;
+import android.content.Context;
+import android.os.Build;
+import android.os.Bundle;
+import android.view.Window;
+import io.sentry.IHub;
+import io.sentry.Integration;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.android.core.gestures.NoOpWindowCallback;
+import io.sentry.android.core.gestures.SentryGestureListener;
+import io.sentry.android.core.gestures.SentryWindowCallback;
+import io.sentry.util.Objects;
+import java.io.Closeable;
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class UserInteractionIntegration
+  implements Integration, Closeable, Application.ActivityLifecycleCallbacks {
+
+  private final @NotNull Application application;
+  private @Nullable IHub hub;
+  private @Nullable SentryAndroidOptions options;
+
+  private final boolean isAllActivityCallbacksAvailable;
+  private final boolean isAndroidXAvailable;
+
+  public UserInteractionIntegration(
+    final @NotNull Application application,
+    final @NotNull IBuildInfoProvider buildInfoProvider,
+    final @NotNull LoadClass classLoader
+  ) {
+    this.application = Objects.requireNonNull(application, "Application is required");
+    Objects.requireNonNull(buildInfoProvider, "BuildInfoProvider is required");
+
+    isAndroidXAvailable = checkAndroidXAvailability(classLoader);
+    isAllActivityCallbacksAvailable =
+      buildInfoProvider.getSdkInfoVersion() >= Build.VERSION_CODES.Q;
```

Comment:
Do we really need this for this integration? that was important for performance since we'd like to start as early as possible, but for the breadcrumbs, maybe not?

Reply:
at first I thought we need this because of this bug https://issuetracker.google.com/issues/186791590 (requires setting a callback before `onCreate`), but I tried it with just onresume/onpause - seems working, will remove